autoload/utils/common.vim

```diff
@@ -18,39 +18,67 @@ endfunction
 " }}} Private functions "
 
 " Executes commands
-function! utils#common#executeCommands(cmds, open_result, ...) abort
-    let l:cwd = get(a:, 1, getcwd())
-    let l:errFormat = get(a:, 2, '')
-
+function! utils#common#executeCommands(cmds, open_result) abort
+    " add all fields
+    let l:commands = []
+    for l:cmd in a:cmds
+        let l:dict_cmd = {'cmd': l:cmd['cmd']}
+        let l:dict_cmd['cwd'] = get( l:cmd, 'cwd', getcwd()  )
+        let l:dict_cmd['errFormat'] = get( l:cmd, 'errFormat', ''  )
+        let l:commands += [l:dict_cmd]
```

Comment:
Now it could be replaced with something like this, just to be more readable.
```
for l:cmd in a:cmds
  call add( l:commands, #{
    \ cmd: l:cmd['cmd'],
    \ cwd: get( l:cmd, 'cwd', getcwd() ),
    \ errFormat: get( l:cmd, 'errFormat', '' )
    \ } )
endfor
```

Reply:
Thanks